src/core/server/saved_objects/migrationsv2/integration_tests/cleanup.test.ts

```diff
@@ -98,7 +98,19 @@ describe('migration v2', () => {
     root = createRoot();
 
     esServer = await startES();
-    await root.setup();
+    const coreSetup = await root.setup();
+
+    coreSetup.savedObjects.registerType({
+      name: 'foo',
+      hidden: false,
+      mappings: {
+        properties: {},
+      },
+      namespaceType: 'agnostic',
```

Comment:
The test suite was triggering a failure by having some `index-pattern` docs registered with the `foo` type

```
{
   type: 'foo',
   id: 'index-pattern:some-id'
}
```

Adding this new unknown doc preflight check was causing the migration to fail too early, failing this test suite.

I'm now registering the `foo` type in the suite, to let it fails at the exact same step it was failing before that PR.